python/core/__init__.py.in

```diff
@@ -447,3 +450,35 @@ QgsRasterTransparency.TransparentThreeValuePixel.percentTransparent = property(g
 QgsRasterTransparency.TransparentThreeValuePixel.percentTransparent = QgsRasterTransparency.TransparentThreeValuePixel.percentTransparent.setter(set_pixel_transparency)
 QgsRasterTransparency.TransparentSingleValuePixel.percentTransparent = property(get_pixel_transparency)
 QgsRasterTransparency.TransparentSingleValuePixel.percentTransparent = QgsRasterTransparency.TransparentSingleValuePixel.percentTransparent.setter(set_pixel_transparency)
+
+
+def qgis_data_type_to_numeric_data_type(dataType: Qgis.DataType) -> Optional[numpy.dtype]:
+   qgis_to_numpy_dtype_dict = {
+    Qgis.DataType.UnknownDataType: None,
+    Qgis.DataType.Byte: numpy.byte,
+    Qgis.DataType.Int8: numpy.int8,
+    Qgis.DataType.UInt16: numpy.uint16,
+    Qgis.DataType.Int16: numpy.int16,
+    Qgis.DataType.UInt32: numpy.uint32,
+    Qgis.DataType.Int32: numpy.int32,
+    Qgis.DataType.Float32: numpy.float32,
+    Qgis.DataType.Float64: numpy.float64,
+    Qgis.DataType.CInt16: numpy.complex64,  # NumPy does not have complex int16, using complex64 as closest alternative
+    Qgis.DataType.CInt32: numpy.complex64,  # NumPy does not have complex int32, using complex64 as closest alternative
+    Qgis.DataType.CFloat32: numpy.complex64,
+    Qgis.DataType.CFloat64: numpy.complex128,
+    Qgis.DataType.ARGB32: numpy.uint32,  # NumPy does not have specific ARGB, using uint32 as closest alternative
```

Comment:
an alternative would be to return a 3D numpy array with the first dimension being A,R,G or B

Reply:
I've set this to None for now. As with the CInt16/32.